### DISCUSSION TITLE
Trouble displaying battery icon in the taskbar

### DISCUSSION THREAD

**Discussion by SpidFightFR** [nix-community/plasma-manager#428] [Q&A] [answered]
Hey again, i tried to enable the display of the battery percentage with:

```nix
programs.plasma {
		widgets = {
		  battery = {
				showPercentage = true;
			};
		};
}
```

But apparently i'm doing something wrong: `widgets` doesn't seem to exist... Should i write battery directly ?

**Comment by HeitorAugustoLN** [comment, marked as answer]
Every widget has their respective configuration contained into where they are defined.
I would assume that you'd want enable in the systemtray. So here it goes:
```nix
programa.plasma = {
  enable = true;
  panels = [
    {
      # Your other panel configurations
      widgets = [
        {
          systemTray = {
            items.configs = {
              battery.showPercentage = true; # This is an high-level widget module, it basically sets what's below:
              "org.kde.plasma.battery".General.showPercentage = true;
            };
          };
        }
      ];
    }
  ];
};
```

Edit: Just read the title

**Reply by HeitorAugustoLN** [reply]
Also, if it is on the desktop, it is basically the same thing, but you'd need to add the size of the widget and it's position. Under `programs.plasma.desktop.widgets`

**Reply by SpidFightFR** [reply]
I tried to apply it but ehhh...
The panel only kept the tray haha:
![image](https://github.com/user-attachments/assets/220b3761-d0c5-44b7-9215-91daec7f35f5)

How could i retrieve something like in the screenshot below ?
![image](https://github.com/user-attachments/assets/c5c69e5e-804b-4d98-b87e-769c546d5db3)

**Reply by SpidFightFR** [reply]
ok i think i got it i'll need some time..

**Reply by HeitorAugustoLN** [reply]
You need to declare every widget in the panel

**Reply by SpidFightFR** [reply]
yep! i'm working on it...

though i have some issues here and there with single widgets...
For example decaring the plasma app-menu (with `widgets.appMenu` in [./modules/widgets/app-menu.nix](https://github.com/nix-community/plasma-manager/blob/trunk/modules/widgets/app-menu.nix)) doesn't seem to work with the modules, i have to declare every widget, one by one, with -for example- `"org.kde.plasma.kickoff"`, for the app menu.

Maybe i'm doing something goofy here, that's possible as well. Imma try again now.

EDIT: content of "org.kde.plasma.<name>" fetched from `$HOME/.config/plasma-org.kde.plasma.desktop-appletsrc`

**Reply by HeitorAugustoLN** [reply]
appmenu is the global menu of plasma. Kickoff is the application launcher, there is a module for it too. Also for icon-tasks, etc.
But yeah, everything is declared in plasma-org.kde.plasma.desktop-appletsrc, so you can just replicate it.

**Reply by SpidFightFR** [reply]
> appmenu is the global menu of plasma. Kickoff is the application launcher, there is a module for it too. Also for icon-tasks, etc. But yeah, everything is declared in plasma-org.kde.plasma.desktop-appletsrc, so you can just replicate it.

Yeah indeed i discovered that to my surprise, turns out "appmenu" is the little menu with "files, options..."
i tried with kicker as well, but it turns out it's more a windows-seven-ish start menu, which is good but kickoff is better ngl.
